src/lib/cfa/ntp_conf.rb

```diff
@@ -346,13 +356,32 @@ def options_matcher
     class CommandRecord < Record
       def options
         return [] unless tree_value?
-        augeas_options.map { |option| option[:key] }
+        augeas_options.each_with_object({}) do |option, opts|
+          opts[option[:key]] = option[:value]
+        end.to_a.flatten.compact
       end
 
       def options=(options)
         ensure_tree_value
         tree_value.tree.delete(options_matcher)
-        options.each { |option| tree_value.tree.add(option, nil) }
+
+        key = nil
+
+        # Some options can be a key value pair or just a single one
+        options.each do |option|
+          if KEY_VALUE_CMD_OPTIONS.include?(option)
+            key = option
+          else
+            add_option(key, option)
+            key = nil
+          end
+        end
```

Comment:
I think this can be problem if value for key-value one is same as some key like `mode ident version 3`. This result in `{ "ident" => "version"}` result.

So maybe code like this
```
i = 0
while i < options.size
  if KEY_VALUE_CMD_OPTIONS.include?(options[i])
    add_options(options[i], options[i] + 1)
    i += 2
  else
   add_options(nil, options[i])
   i += 1
  end
end
```
which I find also a bit more readable.